src/py/_pyodide/_base.py

```diff
@@ -265,6 +267,14 @@ def compile(self) -> "CodeRunner":
             raise AssertionError()
         return self
 
+    def _set_linecache(self):
+        assert self.code
+        filename = self.code.co_filename
+        if filename.startswith("<") and filename.endswith(">"):
+            return
+
+        linecache.cache[filename] = [lambda: self._source]  # type:ignore[assignment]
```

Comment:
I opened a PR to fix the types here:
https://github.com/python/mypy/pull/15650

Reply:
I think that this keep the whole `self` (i.e. the instance of `CodeRunner`) alive forever? Maybe the following is better:
```python
source = self._source
linecache.cache[filename] = [lambda: source]
```

Reply:
Good point, thanks!